src/components/LanguageOption.tsx

```diff
@@ -146,8 +146,22 @@ export default function LanguageOption(props: LanguageOptionProps) {
       .then(() => {
         const metadataExisted = Boolean(metadata?._createdAt)
 
-        callback?.(newTranslationDocument, newMetadataDocument, client)
-
+        try {
```

Comment:
This didn't work when I tested it... I think because it's a promise this should chained:

```
callback?.({
  newTranslationDocument,
  client,
  metadataId,
  sourceLanguageId,
  destinationLanguageId: language.id,
}).catch((err) => {
  toast.push({
    status: 'error',
    title: `Callback`,
    description: `Error while running callback - ${err}.`,
  })
})
```

Reply:
Right you are! My mistake!